flexget/plugins/input/rss.py

```diff
@@ -184,6 +185,23 @@ def process_invalid_content(self, task, data, url):
             f.write(data)
         log.critical('I have saved the invalid content to %s for you to view', filepath)
 
+    def escape_content(self, content):
+        valid_escapes = ('&quot;', '&apos;', '&lt;', '&gt;', '&amp;')
+        future_result = []
+        in_cdata_block = False
+        
+        for idx, char in enumerate(content):
+            char = chr(char)
+            if not in_cdata_block and char == '&':
+                if not str(content[idx:idx+8]).startswith(valid_escapes):
+                    char = '&amp;'
+            elif not in_cdata_block and char == '<' and content[idx:idx+9] == '<![CDATA[':
+                in_cdata_block = True
+            elif in_cdata_block and char == ']' and content[idx-2:idx+1] == ']]>':
```

Comment:
Should be `char == '>'`

Reply:
Maybe it's better to test on ']' because it can be seen in XML not such often as '>'?

Reply:
You're already inside a CDATA block though.

Reply:
But it's common practice to put XML inside CDATA